arrow-cast/src/cast.rs

```diff
@@ -9448,4 +9420,21 @@ mod tests {
         assert_eq!("1969-12-31", string_array.value(1));
         assert_eq!("1989-12-31", string_array.value(2));
     }
+
+    #[test]
+    fn test_nested_list() {
+        let mut list = ListBuilder::new(Int32Builder::new());
+        list.append_value([Some(1), Some(2), Some(3)]);
+        list.append_value([Some(4), Some(5), Some(6)]);
```

Comment:
Can we please include a NULL in the list -- for example, 

```suggestion
        list.append_value([Some(4), Null, Some(6)]);
```